### PR TITLE
Each worker now has its own sticky task queue

### PR DESCRIPTION
**Pull request by Spikhalskiy**
Every worker gets their own sticky task queue and their own fleet of sticky pollers.
Sticky Pollers to normal pollers ratio is fixed at 5:1 in this implementation to reduce the chance for misconfiguration. This ratio may become flexible later or completely go away if the same pollers will switch between sticky a normal queue.

Towards #998

### CODE REVIEW

File: temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java [hunk 1]
```diff
@@ -139,34 +168,40 @@ public boolean isTerminated() {
 
   @Override
   public CompletableFuture<Void> shutdown(ShutdownManager shutdownManager, boolean interruptTasks) {
-    return poller.shutdown(shutdownManager, interruptTasks);
+    return stickyPoller != null
+        ? CompletableFuture.allOf(
+            poller.shutdown(shutdownManager, interruptTasks),
+            stickyPoller.shutdown(shutdownManager, interruptTasks))
+        : poller.shutdown(shutdownManager, interruptTasks);
```
Comment (Sushisource) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java:175]: nit: A lot of side effects happening in one statement, which is also a return statement.

Would read easier I think with some normal conditionals and separate return.

**Review by Sushisource — APPROVED** [review]
Nice, net line removal :)

File: temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java [hunk 2]
```diff
@@ -139,34 +168,40 @@ public boolean isTerminated() {
...
   }
 
   @Override
   public void awaitTermination(long timeout, TimeUnit unit) {
-    poller.awaitTermination(timeout, unit);
+    long timeoutMillis = ShutdownManager.awaitTermination(poller, unit.toMillis(timeout));
+    ShutdownManager.awaitTermination(stickyPoller, timeoutMillis);
```
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java:181]: In other methods you are concerned `stickyPoller` may be `null`, no similar concerns here? Is `awaitTermination` null-safe?

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java [hunk 3]
```diff
@@ -493,12 +503,22 @@ private static SingleWorkerOptions toWorkflowWorkerOptions(
       Scope metricsScope) {
     Map<String, String> tags =
         new ImmutableMap.Builder<String, String>(1).put(MetricsTag.TASK_QUEUE, taskQueue).build();
+
+    Duration stickyQueueScheduleToStartTimeout = options.getStickyQueueScheduleToStartTimeout();
+    if (WorkerOptions.DEFAULT_STICKY_SCHEDULE_TO_START_TIMEOUT.equals(
+            stickyQueueScheduleToStartTimeout)
```
Comment (cretz) [inline temporal-sdk/src/main/java/io/temporal/worker/Worker.java:509]: Didn't read the rest of the code, but assuming this is "if default fallback" code, any concerns of a user explicitly setting the same timeout as our default that we will now override?

**Review by cretz — APPROVED** [review]

File: temporal-sdk/src/main/java/io/temporal/worker/Worker.java (hunk 3 above)
Comment (Spikhalskiy) [reply temporal-sdk/src/main/java/io/temporal/worker/Worker.java:509]: Yeah. For this to "fire", user needs to set an option on the Workers into the default value AND set a DIFFERENT value on WorkerFactory too using the deprecated method. Not a likely scenario to make something special about WorkerOptions#stickyQueueScheduleToStartTimeout and incur maintenance costs in the future.

File: temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java (hunk 2 above)
Comment (Spikhalskiy) [reply temporal-sdk/src/main/java/io/temporal/internal/worker/WorkflowWorker.java:181]: It's null safe, but a good note. I will add a check just for uniformity.

**Pull request merged**